src/backend/env/proposals.rs

```diff
@@ -85,17 +100,8 @@ impl Proposal {
                     return Err("wrong hash".into());
                 }
             }
-            Payload::Fund(receiver, _) => {
-                if Principal::from_text(receiver) == Ok(principal) {
-                    return Err("funding receivers can not vote".into());
-                }
-            }
-            Payload::Reward(Reward {
-                receiver, votes, ..
-            }) => {
-                if Principal::from_text(receiver) == Ok(principal) {
-                    return Err("reward receivers can not vote".into());
-                }
```

Comment:
Not, removed, but moved to the new struct, see line 108.

Good point, I'll add the check for funding as well.

Reply:
[Line 108 in the same file](https://github.com/TaggrNetwork/Taggr/pull/37/files/e66efe53d5af74f85774fecc5682616d2533515d#diff-734e075e3c202adf1c7664a452b15352864a6947f6d021a86b055f759ebc46c9L108)? That's a different validation isn't it? ("reward amount is higher than the configured maximum of {} tokens")

Reply:
Ah, got ya, I was looking at an outdated version of the file (instead of [line 108](https://github.com/TaggrNetwork/Taggr/blob/d9e5475768015d5bac38b5501b59badebcaec85c/src/backend/env/proposals.rs#L108))